stl/inc/expected

```diff
@@ -238,23 +254,8 @@ public:
     // clang-format on
 
     template <class _Uty, class _UErr>
-    static constexpr bool _Allow_unwrapping = disjunction_v<is_same<remove_cv_t<_Ty>, bool>,
-                                                  negation<disjunction<is_constructible<_Ty, expected<_Uty, _UErr>&>, //
-                                                      is_constructible<_Ty, expected<_Uty, _UErr>>, //
-                                                      is_constructible<_Ty, const expected<_Uty, _UErr>&>, //
-                                                      is_constructible<_Ty, const expected<_Uty, _UErr>>, //
-                                                      is_convertible<expected<_Uty, _UErr>&, _Ty>, //
-                                                      is_convertible<expected<_Uty, _UErr>&&, _Ty>, //
-                                                      is_convertible<const expected<_Uty, _UErr>&, _Ty>, //
-                                                      is_convertible<const expected<_Uty, _UErr>&&, _Ty>>>> //
-                                           && !is_constructible_v<unexpected<_Err>, expected<_Uty, _UErr>&> //
-                                           && !is_constructible_v<unexpected<_Err>, expected<_Uty, _UErr>> //
-                                           && !is_constructible_v<unexpected<_Err>, const expected<_Uty, _UErr>&> //
-                                           && !is_constructible_v<unexpected<_Err>, const expected<_Uty, _UErr>>;

```

Comment:
1. I think it's beneficial not to relocate `_Allow_unwrapping` here and in `expected<void, ...>`, as it is used closely after definition so its function can be more obvious.
2. These can also be private:
https://github.com/microsoft/STL/blob/6c69a73911b33892919ec628c0ea5bbf0caf8a6a/stl/inc/expected#L351
https://github.com/microsoft/STL/blob/6c69a73911b33892919ec628c0ea5bbf0caf8a6a/stl/inc/expected#L363